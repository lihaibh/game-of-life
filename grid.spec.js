var { Grid } = require('./grid');

describe("Grid", () => {
    describe("creation", () => {
        it('should fail when providing an invalid seed by the length', () => {
            expect(() => Grid.create(3, 3, "1 0 0 1 1 0")).toThrowErrorMatchingSnapshot();
        });
    });

    describe("neighbors", () => {
        it('should return an array of 8 cell neighbors', () => {
            var grid = Grid.create(3, 3, "1 0 0 1 1 0 0 1 0");

            // out of range
            expect(grid.neighbors(7)).toMatchSnapshot();
            expect(grid.neighbors(-1)).toMatchSnapshot();

            // in the center
            expect(grid.neighbors(4)).toMatchSnapshot();

            // in the edges
            expect(grid.neighbors(0)).toMatchSnapshot();
            expect(grid.neighbors(2)).toMatchSnapshot();
            expect(grid.neighbors(6)).toMatchSnapshot();
            expect(grid.neighbors(8)).toMatchSnapshot();
        });
    });
});