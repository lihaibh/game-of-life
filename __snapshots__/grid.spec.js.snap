// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid creation should fail when providing an invalid seed by the length 1`] = `"the seed is invalid"`;

exports[`Grid neighbors should return an array of 8 cell neighbors 1`] = `
Array [
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  undefined,
  undefined,
]
`;

exports[`Grid neighbors should return an array of 8 cell neighbors 2`] = `
Array [
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
]
`;

exports[`Grid neighbors should return an array of 8 cell neighbors 3`] = `
Array [
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
]
`;

exports[`Grid neighbors should return an array of 8 cell neighbors 4`] = `
Array [
  undefined,
  undefined,
  undefined,
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
]
`;

exports[`Grid neighbors should return an array of 8 cell neighbors 5`] = `
Array [
  undefined,
  undefined,
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
]
`;

exports[`Grid neighbors should return an array of 8 cell neighbors 6`] = `
Array [
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  undefined,
  undefined,
]
`;

exports[`Grid neighbors should return an array of 8 cell neighbors 7`] = `
Array [
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  Object {
    "reducer": [Function],
    "state": Object {
      "alive": true,
    },
  },
  undefined,
  undefined,
  undefined,
  undefined,
]
`;
